src/main/java/com/minecolonies/coremod/colony/buildings/modules/AbstractCraftingBuildingModule.java

```diff
@@ -114,6 +820,26 @@ public boolean isRecipeCompatible(@NotNull final IGenericRecipe recipe)
             return canLearnCraftingRecipes() &&
                     recipe.getIntermediate() == Blocks.AIR;
         }
+
+        @Override
+        public boolean canRecipeBeAdded(@NotNull final IToken<?> token)
+        {
+            if (!super.canRecipeBeAdded(token))
+            {
+                return false;
+            }
+            return isRecipeCompatibleWithCraftingModule(token);
+        }

```

Comment:
The super method already checks `isRecipeCompatibleWithCraftingModule`, so this can be removed?